### PR TITLE
add libpthread to the busybox builder for glibc

### PR DESCRIPTION
**Pull request by puellanivis**

### CODE REVIEW

**Comment by tianon** [comment]
I'm honestly pretty strongly -1 on this addition without further justification such as an official recommendation from BusyBox upstream.  We support three variants of this image (`musl`, `uclibc`, and `glibc`), and this would only increase the disparity between them.

Given that the current justification (as I've understood it) is to allow arbitrary binaries linked against `libpthread` to be dropped in as-is, where do we end up drawing the line?  How long before someone suggests another "common" library that ought to be included for "drop-in arbitrary binaries" support?  Right now, we only include the bare minimum to make BusyBox itself appropriately functional.  IIRC, we don't even include `ca-certificates` or an `ssl_helper` for `wget` to use `https`.

Something that I'd find much more interesting is a way to add these sorts of things easily `FROM busybox` such that they could be added afterwards, which is more likely to lead to general solutions to similar problems such that our users aren't left high and dry if the libraries they need aren't ones we've deemed "common" enough to be included by default.

**Comment by puellanivis** [comment]
Because libpthread is a part of glibc: http://www.sbin.org/doc/glibc/libc_34.html

And the debian libc6 package contains it: https://packages.debian.org/cgi-bin/search_contents.pl?word=libpthread.so.0&searchmode=searchfiles&case=insensitive&version=unstable&arch=i386

And since libpthread is _the_ glibc implementation of the POSIX Threads standard, any binary that uses glibc and threads needs libpthread to be there, (or needs glibc to incorporate libpthread, which is designed to be setup trivially, but debian isn't doing it yet. c.f. libm.so)

Basically _all_ threaded applications using glibc need this. So, if busybox is intended to remain forever a single-thread-supported-only image, then ok, no need to add it.

But if any program wants to use this image and make use of more than one thread, then they would need to wrangle out libpthread for themselves… all while the entire merge request on the image building side is a one line change.

**Comment by bcwaldon** [comment]
I just ran into this issue trying to get a vanilla golang binary running in `busybox:glibc`. Since this has sat for 8 months, is there any chance we will see libpthread incorporated?

**Comment by tianon** [comment]
Thanks for the prod, @bcwaldon -- indeed, this does seem relatively safe in retrospect.  For reference, here's the current size of `libpthread*.so` (~135K):
```
-rwxr-xr-x 1 root root 135K Apr 29 04:23 /lib/x86_64-linux-gnu/libpthread-2.19.so*
```

**Comment by bcwaldon** [comment]
Thanks, @tianon!

**Comment by tianon** [comment]
(https://github.com/docker-library/official-images/pull/2954 is the PR to update this image on the Hub)
